docs/source/our_stack.rst

```diff
@@ -10,31 +10,104 @@ The most up-to-date reference on our codebase is our code itself.
+.com/RoboJackets/rrt>`_ which we use as a submodule. This
+way others can easily use our RRT implementation if they want.
+
+But we can't send a path to a robot and expect it to figure it out. We need
+to give our firmware a single velocity command for that instant of planning.
+The file ``soccer/src/soccer/planning/primitives/velocity_profiling.cpp``
+finds that velocity trajectory which we can then return as the current
+trajectory as the robot.
+
+At the highest level of planning, we have several different planners which the
+planning_node selects based on the skill that gameplay requested. They
+have different behaviors and based on the their names which give you insight on their purpose. They are all located
+in ``soccer/src/soccer/planning/planner``.
+
+Although this is not an extensive explanation of everything planning does, it is a good start, and looking
+through the ``planning`` folder will give you more information about this topic.
+
+Processor
+---------
+It is defined in ``soccer/src/soccer/processor.cpp``, and it
+starts multiple temp nodes. It then updates the context in an infinite loop. Context is used to update the ui. It is good that most other nodes
+are not dependent on this loop.
 
 Radio
 -----
+See the ``/radio`` directory. We have a sim radio for when using virtual
+matches or testing locally, and a network radio for when we run on real
+hardware. It sends commands to our robots using the league designated ports
+for yellow and blue when in sim mode. The simulation packets are standardized
+by the league and in the ``rj_protos`` directory (the ones with the ssl
+prefix).
+
+The network radio expects each robot to know the ip address of where to send
+to. When soccer eventually receives a robot status message from a robot, it
+adds the robot id to an ip map. Now our software knows how to reach that
+robot id. If that robot doesn't send any robot status in a while, it will
+disconnect and disappear from the UI (when running with real robots).
 
 Referee
 -------
+Referee controls the match state. We have an internal referee which can send
+quick commands like stop and start from our UI, so we don't always have to
+launch the game controller. When we do want to use the game controller
+though, the external referee handles that by listening for game events and
+updates on specific ports designated by the league.
 
 UI
 --
+We use Qt for UI development. Code for it is located in ``soccer/src/soccer/ui``.
+The ``qt`` directory contains the view as .ui files and images that are
+placed on the view.
+You could directly edit these .ui files based on their
+proper syntax to change our view, but I wouldn't recommend it.
+Instead install Qt designer and open .ui files through it to save yourself a
+lot of time.
+
+After the view, all the C++ files act as the view-model for each view.
+Some of them are quite messy and do not follow our naming conventions; however,
+ if you follow the common method patterns, you should be able to
```

Comment:
this line renders weirdly
<img width="704" alt="Screen Shot 2022-07-17 at 5 37 43 PM" src="https://user-images.githubusercontent.com/30275369/179425703-8a10ead1-2d48-4e88-8513-cb0283560825.png">